game/games/snake/canvas.gd

```diff
@@ -0,0 +1,166 @@
+extends Node2D
+
+enum Direction { UP, DOWN, LEFT, RIGHT }
+
+export var tile_count: int = 22
+export var white_space: float = 80.0
+
+var _tile_size: int = 0
+
+var _snake: Array = []
+var _direction = Direction.RIGHT
+var _apple_pos := Vector2()
+
+var _score: int = 0
+var _high_score: int = 0
+
+var _tick: int = 0
+var _game_paused: bool = false
+
+onready var _main = get_parent().get_parent().get_parent().get_parent()
+
+
+func _ready():
+	_set_game_paused(true)
+	randomize()
+	_calc_tile_size()
+	_set_playfield()
+	_set_snake()
+	_set_apple()
+
+
+func _process(_delta):
+	_tick += 1
+	if _tick == 4:
+		_tick = 0
+		var new_head_pos = _move_snake()
+		var collision = _check_for_collision(new_head_pos)
+		if collision:
+			_game_over()
+		else:
+			_check_for_apple(new_head_pos)
+		update()  # triggers NOTIFICATION_DRAW of CanvasItem / _draw()
+
+
+func _draw():
+	_draw_snake()
+	_draw_apple()
+
+
+func _input(_event):
+	if Input.is_action_just_pressed("ui_up"):
+		_set_game_paused(false)
+		if _direction != Direction.DOWN:
+			_direction = Direction.UP
+	elif Input.is_action_just_pressed("ui_down"):
+		_set_game_paused(false)
+		if _direction != Direction.UP:
+			_direction = Direction.DOWN
+	elif Input.is_action_just_pressed("ui_left"):
+		_set_game_paused(false)
+		if _direction != Direction.RIGHT:
+			_direction = Direction.LEFT
+	elif Input.is_action_just_pressed("ui_right"):
+		_set_game_paused(false)
+		if _direction != Direction.LEFT:
+			_direction = Direction.RIGHT
+
+
+func _calc_tile_size():
+	var spacing = Vector2(white_space * 2, white_space * 2)
+	var slice = (OS.get_window_size() - spacing) / tile_count
+	var size = 0
+	if slice.x > slice.y:
+		size = slice.y
+	else:
+		size = slice.x
+	_tile_size = int(floor(size))
+
+
+func _set_playfield():
+	var size = _tile_size * tile_count
+	get_parent().rect_min_size = Vector2(size, size)
+
+
+func _set_game_paused(value: bool):
+	set_process(not value)
+	_game_paused = value
+
+
+func _set_snake():
+	_snake.clear()
+	var center = int(tile_count / 2)
+	_snake.push_back(Vector2(center, center))
+	_direction = null
+
+
+func _set_apple():
```

Comment:
What if the apple spawns inside the snake?

Reply:
then it is drawn on top on the snake